cmd/github-post: TestListFailures failed

cmd/github-post.TestListFailures [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7785166?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7785166?buildTab=artifacts#/) on release-22.2 @ [ca673993473629a46370100108dd8c185a592e91](https://github.com/cockroachdb/cockroach/commits/ca673993473629a46370100108dd8c185a592e91):


```
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestXXA in pkg: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
I221201 08:32:50.299410 216 (gostd) main.go:550  [-] 34  errror getting file:line for github.com/cockroachdb/cockroach/pkg/kv.TestXXA: could not find test TestXXA
I221201 08:32:50.299618 216 (gostd) main.go:555  [-] 35  loading codeowners: TEAMS.yaml not found
I221201 08:32:50.299676 216 (gostd) main.go:442  [-] 36  timeout culprit found: TestXXX/sub3
couldn't find test TestXXX in pkg/kv: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestXXX in pkg: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
I221201 08:32:50.307125 216 (gostd) main.go:550  [-] 37  errror getting file:line for github.com/cockroachdb/cockroach/pkg/kv.TestXXX/sub3: could not find test TestXXX
I221201 08:32:50.307303 216 (gostd) main.go:555  [-] 38  loading codeowners: TEAMS.yaml not found
    main_test.go:306: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/451/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:306
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/451/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main.go:443
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/451/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:312
        	Error:      	Not equal: 
        	            	expected: []string{"@cockroachdb/test-eng"}
        	            	actual  : []string(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=21) "@cockroachdb/test-eng"
        	            	-}
        	            	+([]string) <nil>
        	            	 
        	Test:       	TestListFailures/timeout-culprit-not-found.json
    main_test.go:307: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/451/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:307
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/451/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main.go:443
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/451/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:312
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestListFailures/timeout-culprit-not-found.json
    --- FAIL: TestListFailures/timeout-culprit-not-found.json (0.03s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListFailures.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21984

cmd/github-post.TestListFailures [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7785170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7785170?buildTab=artifacts#/) on release-22.2 @ [ca673993473629a46370100108dd8c185a592e91](https://github.com/cockroachdb/cockroach/commits/ca673993473629a46370100108dd8c185a592e91):


```
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/72/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main.go:387
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/72/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:312
        	Error:      	Not equal: 
        	            	expected: []string{"@cockroachdb/kv"}
        	            	actual  : []string(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=15) "@cockroachdb/kv"
        	            	-}
        	            	+([]string) <nil>
        	            	 
        	Test:       	TestListFailures/timeout-culprit-found.json
I221201 08:32:57.003994 68 (gostd) main.go:409  [-] 25  failed to create slow tests report: open artifacts/slow-tests-report.txt: no such file or directory
    main_test.go:307: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/72/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:307
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/72/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main.go:387
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/72/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:312
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestListFailures/timeout-culprit-found.json
couldn't find test TestAbortReadOnlyTransaction in pkg/kv/kvclient/kvcoord: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestAbortReadOnlyTransaction in pkg/kv/kvclient: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestAbortReadOnlyTransaction in pkg/kv: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestAbortReadOnlyTransaction in pkg: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
I221201 08:32:57.016242 68 (gostd) main.go:550  [-] 26  errror getting file:line for github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.TestAbortReadOnlyTransaction: could not find test TestAbortReadOnlyTransaction
I221201 08:32:57.016954 68 (gostd) main.go:555  [-] 27  loading codeowners: TEAMS.yaml not found
I221201 08:32:57.017049 68 (gostd) main.go:442  [-] 28  timeout culprit found: TestAbortReadOnlyTransaction
    main_test.go:316: expected 2 issues, got: 1
    --- FAIL: TestListFailures/timeout-culprit-found.json (0.03s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListFailures.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cmd/github-post.TestListFailures [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7785162?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7785162?buildTab=artifacts#/) on release-22.2 @ [ca673993473629a46370100108dd8c185a592e91](https://github.com/cockroachdb/cockroach/commits/ca673993473629a46370100108dd8c185a592e91):


```
        	Test:       	TestListFailures/stress-failure.json
    main_test.go:307: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/908/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:307
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/908/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main.go:387
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/908/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:312
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestListFailures/stress-failure.json
    --- FAIL: TestListFailures/stress-failure.json (0.02s)
=== RUN   TestListFailures/stress-subtests.json
I221201 08:33:31.498077 22 (gostd) main.go:363  [-] 16  consolidating failed subtest "1.0" into parent test "TestPretty"
I221201 08:33:31.498167 22 (gostd) main.go:363  [-] 17  consolidating failed subtest "[\"hello\",_[\"world\"]]" into parent test "TestPretty"
I221201 08:33:31.498191 22 (gostd) main.go:367  [-] 18  failed parent test "TestPretty" (no subtests)
couldn't find test TestPretty in pkg/util/json: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestPretty in pkg/util: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
couldn't find test TestPretty in pkg: fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 exit status 128
I221201 08:33:31.513050 22 (gostd) main.go:550  [-] 19  errror getting file:line for github.com/cockroachdb/cockroach/pkg/util/json.TestPretty: could not find test TestPretty
I221201 08:33:31.513232 22 (gostd) main.go:555  [-] 20  loading codeowners: TEAMS.yaml not found
    main_test.go:306: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/908/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:306
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/908/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main.go:387
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/908/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cmd/github-post/github-post_test_/github-post_test.runfiles/com_github_cockroachdb_cockroach/pkg/cmd/github-post/main_test.go:312
        	Error:      	Not equal: 
        	            	expected: []string{"@cockroachdb/unowned"}
        	            	actual  : []string(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=20) "@cockroachdb/unowned"
        	            	-}
        	            	+([]string) <nil>
        	            	 
        	Test:       	TestListFailures/stress-subtests.json
    --- FAIL: TestListFailures/stress-subtests.json (0.02s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListFailures.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/build/teamcity/cockroach/nightlies/stress_impl.sh
+++ b/build/teamcity/cockroach/nightlies/stress_impl.sh
@@ -16,6 +16,17 @@ bazel build //pkg/cmd/bazci //pkg/cmd/github-post //pkg/cmd/testfilter --config=
 BAZEL_BIN=$(bazel info bazel-bin --config=ci)
 ARTIFACTS_DIR=/artifacts
 
+if [[ ! -z $(bazel query "attr(tags, \"broken_in_bazel\", $TARGET)") ]]
+then
+    echo "Skipping test $TARGET as it is broken in bazel"
+    exit 0
+fi
+if [[ ! -z $(bazel query "attr(tags, \"integration\", $TARGET)") ]]
+then
+    echo "Skipping test $TARGET as it is an integration test"
+    exit 0
+fi
+
 GOTESTTIMEOUTSECS=$(($TESTTIMEOUTSECS - 5))
 GO_TEST_JSON_OUTPUT_FILE=$ARTIFACTS_DIR/$(echo "$TARGET" | cut -d: -f2).test.json.txt
 exit_status=0

